src/ChangeStream.php

```diff
@@ -72,7 +74,16 @@ public function next()
                 $this->key++;
             }
         } catch (RuntimeException $e) {
-            $this->resume();
+            $resumable = false;
+            if (strpos($e->getMessage(), "not master") !== false) {
+                $resumable = true;
+            }
+            if ($e->getCode() === self::CURSOR_NOT_FOUND) {
+                $resumable = true;
+            }
```

Comment:
[ConnectionException](http://php.net/manual/en/class.mongodb-driver-exception-connectionexception.php) is a sub-class of RuntimeException, so it would enter this try/catch. Can we add one more check for an instance of that class here to flip on `$resumable`?

[`phongo_exception_from_mongoc_domain()`](https://github.com/mongodb/mongo-php-driver/blob/1.3.3/php_phongo.c#L138) in the driver has all of the logic for converting libmongoc error codes into PHP exception classes.